src/main/java/moe/yushi/authlibinjector/APIMetadata.java

```diff
@@ -54,24 +58,33 @@ public static APIMetadata parse(String apiRoot, String metadataResponse) throws
 						.map(JsonUtils::asJsonString)
 						.map(KeyUtils::parseSignaturePublicKey);
 
+		Set<PublicKey> decodedPublickeys =
+				ofNullable(response.get("signaturePublickeys"))
+						.map(it -> asJsonArray(it).stream()
+								.map(JsonUtils::asJsonString)
+								.map(KeyUtils::parseSignaturePublicKey)
+								.collect(toSet()))
+						.orElse(emptySet());
```

Comment:
`emptySet()` returns an immutable set, causing `decodedPublickeys::add` to throw an `UnsupportedOperationException`. Should be `.orElseGet(HashSet::new)` here.